test/ContractTestContext.ts

```diff
@@ -12,85 +24,96 @@ async function getContractFactories (factoryNames: string[]): Promise<{ [K: stri
     }), {});
 }
 
-export class ContractTestContext {
-    // addresses for different roles
-    owner;
-    addr1;
-    addr2;
-    addr3;
-    treasurySigner;
-
-    // Archimedes contracts
-    coordinator;
-    cdp;
-    vault;
-    lvUSD;
-    exchanger;
-    leverageEngine;
-    leverageAllocator;
-    positionToken;
-    parameterStore;
-
+export type ContractTestContext = {
+    coordinator: Coordinator;
+    owner: SignerWithAddress;
+    addr1: SignerWithAddress;
+    addr2: SignerWithAddress;
+    addr3: SignerWithAddress;
+    treasurySigner: SignerWithAddress;
+    // Archimedes deployed contract instances
+    cdp: CDPosition;
+    vault: VaultOUSD;
+    lvUSD: LvUSDToken;
+    exchanger: Exchanger;
+    leverageEngine: LeverageEngine;
+    leverageAllocator: LeverageAllocator;
+    positionToken: PositionToken;
+    parameterStore: ParameterStore;
     // External contracts
-    externalOUSD;
+    externalOUSD: Contract;
+}
+
+export async function buildContractTestContext (): Promise<ContractTestContext> {
```

Comment:
It's a bit of a pain getting the class types to work nicely when an async method like setup is required. Instead just converting to an async function that returns the context as an object.